File: python/leetcode/66-plus-one.py
from collections import deque


def plusOne(digits):
    new_digits = deque()
    carry = 0

    for i in range(len(digits)-1, -1, -1):
        result = carry
        result += digits[i]
        if i == len(digits) - 1:
            result += 1
        new_digits.appendleft(result % 10)
        carry = (result // 10)
        print('result', result)
        print('carry', carry)

    if carry:
        new_digits.appendleft(carry)

    return new_digits


print(plusOne([9]))

File: c#/leetcode/66-plus-one.cs
public class Solution
{
  public int[] PlusOne(int[] digits)
  {
    int carry = 0;
    List<int> d = new List<int>(digits);

    for (int i = d.Count - 1; i >= 0; i--)
    {
      if (i == d.Count - 1)
      {
        d[i]++;
      }
      if (carry > 0)
      {
        d[i] += carry;
      }
      carry = d[i] / 10;
      d[i] = d[i] % 10;
    }

    if (carry > 0)
    {
      d.Insert(0, carry);
    }





    return d.ToArray();
  }
}